src/kv-installation-store.ts

```diff
@@ -152,34 +240,87 @@ export class KVInstallationStore<E extends SlackOAuthEnv> implements Installatio
           userScopes: user?.user_scopes,
         };
       } catch (e) {
-        throw new AuthorizeError(`Failed to authorize (error: ${e}, query: ${JSON.stringify(query)})`);
+        throw new AuthorizeError(
+          `Failed to authorize (error: ${e}, query: ${JSON.stringify(query)})`
+        );
       }
     };
   }
+
+  /**
+   * Calls the `auth.test` Slack API method, first checking the auth.test cache if enabled.
+   * @param client - The Slack API client to use for the request.
+   * @param token - The token to use for the request, and/or cache key.
+   * @returns The response from the `auth.test` Slack API method.
+   */
+  async callAuthTest(
+    client: SlackAPIClient,
+    token: string | undefined
+  ): Promise<AuthTestResponse> {
+    if (token && this.#authTestCacheEnabled && this.#authTestCacheNamespace) {
+      const cachedResponse = await this.#authTestCacheNamespace.get(token);
+      if (cachedResponse) {
+        return JSON.parse(cachedResponse);
+      }
+
+      const authTestResponse = await client.auth.test();
+      const permanentCacheEnabled = this.#authTestCacheExpirationSecs <= 0;
+      await this.#authTestCacheNamespace.put(
+        token,
+        JSON.stringify(authTestResponse),
```

Comment:
The HTTP Headers object is not serializable, so this simple cache loses the headers value. We need to introduce a new CachedSlackAPIResponse type with a parsed record object for headers, and save/load such instead.

Reply:
the reason why we need to support headers here is https://github.com/slack-edge/slack-cloudflare-workers/blob/1.3.2/src/kv-installation-store.ts#L108

Reply:
Great catch